src/IntercomCompanies.php

```diff
@@ -32,6 +32,19 @@ public function update($options)
     {
         return $this->create($options);
     }
+    
+    /**
+     * Deletes a Company.
+     *
+     * @see https://developers.intercom.com/intercom-api-reference/reference#delete-a-company
+     * @param array $options
+     * @return stdClass
+     * @throws Exception
+     */
+    public function delete($id, $options = [])
+    {
+        return $this->client->delete($this->companyPath($id), $options);
```

Comment:
Minor sugggestion, mainly cause it seems like a pattern followed in other resources.
```suggestion
        $path = $this->companyPath($id);
        return $this->client->delete($path, $options);
```